lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -127,14 +134,19 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError as e:
             self.logger.error(f"Indices from metadata table not found {e!r}")
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+            raise InvalidAPIRequest(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                f"Indices from metadata table not found {e!r}",
+            )
 
         if index_map is None:
             self.logger.error(
                 f"server.index-map not found in Metadata for a dataset {dataset!r}"
             )
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-
+            raise InvalidAPIRequest(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                f"server.index-map not found in Metadata for a dataset {dataset!r}",
+            )
```

Comment:
Not your fault since you're using the existing text here; but, really, the `MetadataError` handler up above is really "server.index-map key not found"; *this* case is "the key was found but has no useful value".

And while the text of these `logger.error` messages is OK because they're for our own internal consumption when analyzing a server problem, we shouldn't be confusing the external messages with this sort of internal detail.

Our convention has been that for "internal server errors" (where we really don't understand the exact cause in most cases), we've just said "INTERNAL ERROR" (leaving it to a developer or server admin to analyze the log journal); this message is too detailed to be useful, because most users won't have any idea what that means. This means that somehow either our database has become corrupt, or something went wrong when we indexed the dataset and we didn't catch it at the time; it's not a problem with the client's API query.

I think that the `self.logger.error` messages for these two cases (here and line 136) could be reversed, which would be a more accurate description; and the `InvalidAPIRequest` exceptions should go back to "INTERNAL ERROR".

Reply:
> Our convention has been that for "internal server errors" (where we really don't understand the exact cause in most cases), we've just said "INTERNAL ERROR" (leaving it to a developer or server admin to analyze the log journal)

I'd like to push back on this just a little bit.

Our convention for returning vague messages (e.g., `"INTERNAL ERROR"`) has mostly applied to security-related failures.

I have no problems with us returning detailed messages to the requestor (provided the contents are not security-sensitive), so long as it's under a `500`/`INTERNAL_ERROR` status.  That status indicates that the fault lies on the server side and that the requestor is not expected to be able to address it.  **_But_**, if we give a good message, then the requestor can relay that to us if it happens, and we might not _need_ to dive into the logs...which I think is worthwhile.

I don't think that any message we give under a `500` status will make the situation worse.  I don't see how it would  make the message "confusing" for the requestor or user.